Formula/openssl@1.1.rb

```diff
@@ -74,23 +93,44 @@ def install
     # Whilst our env points to opt_bin, by default OpenSSL resolves the symlink.
     ENV["PERL"] = Formula["perl"].opt_bin/"perl" if which("perl") == Formula["perl"].opt_bin/"perl"
 
-    arch_args = %w[darwin64-x86_64-cc enable-ec_nistp_64_gcc_128]
+    arch_args = []
+    on_macos do
+      arch_args = %w[darwin64-x86_64-cc enable-ec_nistp_64_gcc_128]
+    end
+    on_linux do
+      if Hardware::CPU.intel?
+        arch_args << (Hardware::CPU.is_64_bit? ? "linux-x86_64" : "linux-elf")
+      elsif Hardware::CPU.arm?
+        arch_args << (Hardware::CPU.is_64_bit? ? "linux-aarch64" : "linux-armv4")
+      end
+    end
     # Remove `no-asm` workaround when upstream releases a fix
     # See also: https://github.com/openssl/openssl/issues/12254
     arch_args << "no-asm" if Hardware::CPU.arm?
 
     ENV.deparallelize
     system "perl", "./Configure", *(configure_args + arch_args)
     system "make"
-    system "make", "test"
+    on_macos do
+      system "make", "test"
+    end
     system "make", "install", "MANDIR=#{man}", "MANSUFFIX=ssl"
+    # See https://github.com/Linuxbrew/homebrew-core/pull/8891
+    on_linux do
+      system "make", "test"
```

Comment:
How about moving `make test` after `make install` for both macOS and Linux? I guess it won't make a difference on macOS.